src/elements/Svg.tsx

```diff
@@ -81,16 +86,46 @@ export default class Svg extends Shape<SvgProps> {
     root && root.setNativeProps(props);
   };
 
-  toDataURL = (callback: (base64: string) => void, options?: object) => {
+  /**
+   * @deprecated use size property instead.
+   * @description Converts the SVG component to a data URL in either JPEG or PNG format.
+   * @param {function(string): void} callback - A function that receives the base64-encoded data URL string.
+   * @param {DataUrlOptions} options - The options to specify the format (JPEG or PNG) and additional parameters.
+   */
+  toDataURL(callback: (base64: string) => void, options: Size): void;
+  toDataURL(callback: (base64: string) => void): void;
+  toDataURL(callback: (base64: string) => void, options: DataUrlOptions): void;
+  toDataURL(
+    callback: (base64: string) => void,
+    options?: DataUrlOptions | Size
+  ): void {
     if (!callback) {
       return;
     }
+    if (options && 'format' in options) {
+      validateOptions(options);
+    }
     const handle = findNodeHandle(this.root as Component);
     const RNSVGSvgViewModule: Spec =
       // eslint-disable-next-line @typescript-eslint/no-var-requires
       require('../fabric/NativeSvgViewModule').default;
     RNSVGSvgViewModule.toDataURL(handle, options, callback);
-  };
+  }
+
+  /**
+   * Options specific to JPEG format.
```

Comment:
> to JPEG format

Doesn't this apply to PNG as well? I think this description is redundant and could be removed.